### PR TITLE
pre-build images during acceptance testing

### PR DESCRIPTION
**Pull request by sadlerap**
This allows for primaza's acceptance test suite to run against pre-built images, controlled with a set of environment variables.  See `TESTING.md` for documentation.

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Acceptance tests can be controlled with a few key environment variables:
+
+- `PRIMAZA_CONTROLLER_IMAGE_REF`: use the image provided as the primaza controller.
+- `PRIMAZA_AGENTAPP_IMAGE_REF`: use the image provided as the application agent controller.
+- `PRIMAZA_AGENTSVC_IMAGE_REF`: use the image provided as the service agent controller.
+- `PULL_IMAGES`: due to technical restrictions related to our use of `kind`,
+  images need to be in the local docker registry in order to be tested.  To
+  ensure these images exist locally, set this env var to have the test runner
+  pull these images before running test scenarios.
```
Comment (dperaza4dustbit) [inline TESTING.md:38]: Will it be worth describing the defaults if these variables are not set?

File: make/acceptance-tests.mk [hunk 2]
```diff
@@ -11,6 +11,53 @@ else
...
+
+ifeq ($(origin PRIMAZA_CONTROLLER_IMAGE_REF), undefined)
+export PRIMAZA_CONTROLLER_IMAGE_REF = primaza-controller:latest
+endif
+ifeq ($(origin PRIMAZA_AGENTAPP_IMAGE_REF), undefined)
+export PRIMAZA_AGENTAPP_IMAGE_REF = agentapp:latest
+endif
+ifeq ($(origin PRIMAZA_AGENTSVC_IMAGE_REF), undefined)
+export PRIMAZA_AGENTSVC_IMAGE_REF ?= agentsvc:latest
+endif
```
Comment (dperaza4dustbit) [inline make/acceptance-tests.mk:26]: These will be the defaults you can use in docs

File: test/acceptance/features/steps/kind.py [hunk 3]
```diff
@@ -91,28 +92,26 @@ def ipaddress(self):
 
 
 class PrimazaKind(PrimazaCluster):
+    __controller_image: str = get_env("PRIMAZA_CONTROLLER_IMAGE_REF")
+    __agentapp_image: str = get_env("PRIMAZA_AGENTAPP_IMAGE_REF")
+    __agentsvc_image: str = get_env("PRIMAZA_AGENTSVC_IMAGE_REF")
+
```
Comment (dperaza4dustbit) [inline test/acceptance/features/steps/kind.py:98]: error handling? what if the env var is not set? Maybe set defaults in the get_env call here as well.

**Review by dperaza4dustbit — COMMENTED** [review]
This PR looks good, just a few suggestions

File: TESTING.md (hunk 1 above)
Comment (sadlerap) [reply TESTING.md:38]: Yes, it's definitely useful to call out default behavior.

Comment (sadlerap) [reply TESTING.md:38]: Fixed in 112f4d2.

File: make/acceptance-tests.mk (hunk 2 above)
Comment (sadlerap) [reply make/acceptance-tests.mk:26]: Fixed in 112f4d2.

File: test/acceptance/features/steps/kind.py (hunk 3 above)
Comment (dperaza4dustbit) [reply test/acceptance/features/steps/kind.py:98]: OK

**Review by dperaza4dustbit — APPROVED** [review]
lgtm

**Pull request merged**